packages/itwinui-react/src/core/LabeledInput/LabeledInput.tsx

```diff
@@ -83,29 +99,49 @@ export const LabeledInput = React.forwardRef((props, ref) => {
   const icon = svgIcon ?? (status && StatusIconMap[status]());
 
   return (
-    <InputContainer
-      label={label}
-      disabled={disabled}
-      required={required}
-      status={status}
-      message={message}
-      icon={icon}
-      isLabelInline={displayStyle === 'inline'}
-      isIconInline={iconDisplayStyle === 'inline'}
-      className={className}
-      style={style}
-      inputId={id}
-    >
-      <Input
-        disabled={disabled}
-        className={inputClassName}
-        style={inputStyle}
-        required={required}
-        ref={ref}
-        id={id}
-        {...rest}
-      />
-    </InputContainer>
+    <InputGrid labelPlacement={displayStyle} {...wrapperProps}>
+      {label && (
+        <Label
+          as='label'
+          required={required}
+          disabled={disabled}
+          htmlFor={id}
+          {...labelProps}
+        >
+          {label}
+        </Label>
+      )}
+
+      <InputWithDecorations
+        status={status}
+        isDisabled={disabled}
+        {...inputWrapperProps}
+      >
+        <InputWithDecorations.Input
+          disabled={disabled}
+          required={required}
+          id={id}
+          ref={ref}
+          {...rest}
+        />
+        {icon && iconDisplayStyle === 'inline' && (
+          <Icon fill={!svgIcon ? status : undefined} padded {...iconProps}>
+            {icon}
+          </Icon>
+        )}
+      </InputWithDecorations>
+
+      {(message || (icon && iconDisplayStyle !== 'inline')) && (
```

Comment:
i think there is something missing now. previously there was the ability to manually pass a `<StatusMessage>` which will now break

(same comment in `LabeledTextarea` and `LabeledSelect`)

Reply:
Added a check to see if message is not a string - render full message